src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -629,5 +688,45 @@ void get_num_streams(const int streams, const std::shared_ptr<ov::Model>& model,
     generate_stream_info(streams, -1, model, config, proc_type_table);
 }
 
+void update_stream_with_weights(const std::shared_ptr<const ov::Model>& model, Config& config) {
+    if (config.numSubStreams > 0) {
+        bool has_large_fc = false;
+        for (const auto& op : model->get_ops()) {
+            if (!ov::is_type<ov::intel_cpu::FullyConnectedNode>(op))
+                continue;
+
+            auto weights = op->input_value(1);
+            if (!ov::op::util::is_on_constant_path(weights))
+                continue;
+
+            const auto element_num = ov::shape_size(weights.get_shape());
+            if (element_num <= 6600000)
```

Comment:
Please isolate this magic constant and add a proper description regarding the heuristic.

Reply:
Based on discussion in Arch Forum review,  default setting for TP remains disabled. So this heuristic logic will be removed in this PR.